.circleci/config.yml

```diff
@@ -16,11 +16,19 @@ jobs:
           keys:
             - mix-cache-{{ checksum "mix.lock" }}
 
+      - restore_cache:
+          keys:
+            - dialyzer-cache-21.2-1.8.1
+
       - run: mix do deps.get
       - save_cache:
           key: mix-cache-{{ checksum "mix.lock" }}
           paths: "deps"
 
       - run: mix format --check-formatted
       - run: mix test
+      - run: mix dialyzer
+      - save_cache:
+          key: dialyzer-cache-21.2-1.8.1
```

Comment:
Do you know if it's possible to use the `checksum` here so that the cache is busted when the dependencies change? That way we don't have to manually update this when we update dependencies. 

We could use the `".tool-versions"` file, since that's the source of truth for Erlang and Elixir versions. So if that file changes, it means we should bust this cache. 

What do you think? Would it be possible to do this? 

```
- save_cache: 
    key: dialyzer-cache-{{ checksum ".tool-versions" }}
```

And the corresponding change in the `-restore_cache` entry?